src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DocumentMapping/RazorEditHelper.cs

```diff
@@ -61,6 +60,35 @@ internal static async Task<ImmutableArray<RazorTextChange>> MapCSharpEditsAsync(
         return NormalizeEdits(textChangeBuilder.DrainToOrderedImmutable(), telemetryReporter, cancellationToken);
     }
 
+    private static async Task<ImmutableArray<string>> GetUsingsDirectivesAsync(SyntaxTree originalSyntaxTree, CancellationToken cancellationToken)
+    {
+        var syntaxRoot = await originalSyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+        return syntaxRoot
+            .DescendantNodes(n => n is BaseNamespaceDeclarationSyntax or CompilationUnitSyntax)
+            .OfType<UsingDirectiveSyntax>()
+            .SelectAsArray(GetNamespaceFromDirective);
+    }
+
+    private static string GetNamespaceFromDirective(UsingDirectiveSyntax usingDirectiveSyntax)
+    {
+        var str = usingDirectiveSyntax.ToFullString();
```

Comment:
Random code written without a compiler, but rather than realising the string here and trimming twice, is it more efficient to do something like this:

```
// .Span ignores trivia, so by the rules of Roslyn it won't contain whitespace, right??
var span = usingDirectiveSyntax.Span;
if (!usingDirectiveSyntax.SemicolonToken.IsMissing)
{
	span = new TextSpan(span.Start, span.Length -1);
}

if (!usingDirectiveSyntax.UsingKeyword.IsMissing)
{
	span = ....
}

return sourceText.GetSubTextString(span);
```

Reply:
💯

Reply:
Honestly I got so in the habit of how the razor syntax tree looks I forgot about sensible things like the span not including trivia. This can be simplified even more by making some assumptions (like explicitly only handling things where the `Name` is not null,) 

```csharp
private static string GetNamespaceFromDirective(UsingDirectiveSyntax usingDirectiveSyntax, SourceText sourceText)
{
    var nameSyntax = usingDirectiveSyntax.Name.AssumeNotNull();

    if (usingDirectiveSyntax.Alias is null)
    {
        return sourceText.GetSubTextString(nameSyntax.Span);
    }

    return sourceText.GetSubTextString(TextSpan.FromBounds(usingDirectiveSyntax.Alias.Span.Start, nameSyntax.Span.End));
}
```

Reply:
`using static`?

No idea if it's possible in a razor file, but worth adding a test for, either to validate or prove it's not supported?

Reply:
I'm guessing since it's just a csharp block that it is supported. I'll add a test and I think I can just find the leftmost ChildNode since the using keyword is a token.

Reply:
Scratch that, static is a token. I guess it's left most child nodes and tokens excluding the using keyword

Reply:
What about the start position just being the using keywords span end, plus end trivia length?

Reply:
Is it guarantee that the end trivia is attached to the using keyword? That's the only thing I wasn't sure of. I'm rusty on my Roslyn syntax

Reply:
Guaranteed, no idea. From a quick test, it was.